### PR TITLE
feat(CLI): initial project setup

### PR DESCRIPTION
**Pull request by frdomovic**
# feat(CLI): initial project setup

## Summary:
Initial template of CLI function

## Tests:

https://github.com/calimero-is-near/cali2.0-experimental/assets/93442516/f514fee9-9dc8-4cf2-a15f-ee367bf3f29c

### CODE REVIEW

File: client/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env node
+import yargs from "yargs";
+import { hideBin } from "yargs/helpers";
+import { JoinCommandOptions, StartSessionCommandOptions, AddKeysCommandOptions } from "./types";
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const chalk = require("chalk");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const fs = require("fs");
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const open = require("open");
```
Comment (xilosada) [inline client/src/index.ts]: Why not import?

File: client/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "es2016",
+    "module": "commonjs",
```
Comment (frdomovic) [inline client/tsconfig.json]: update

File: client/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "target": "es2016",
```
Comment (frdomovic) [inline client/tsconfig.json]: update

File: client/src/index.ts (hunk 1 above)
Comment (frdomovic) [reply client/src/index.ts]: Updated to import, this was tsconfig issue but not its updated to esnext

**Pull request merged**